### DISCUSSION TITLE
Receiving " Firebase: No Firebase App '[DEFAULT]' has been created", even though it has been created

### DISCUSSION THREAD

**Discussion by vpaladino778** [gladly-team/next-firebase-auth#373] [Q&A] [answered]
I'm receiving the following error on my project, even though not much is different from the example project. 

One thing that I made a change to was the import statement in `login.tsx`. In the example, they import `firebase/app`, but I had to import `firebase/compact/app` (I was getting `Property 'auth' does not exist on type` with the former)

Does anyone have any ideas or suggestions? Maybe I'm missing something, i figured since initAuth is being called in _app.tsx, then firebase should be initialized. Could this be because I'm importing firebase/app/compact?

![image](https://user-images.githubusercontent.com/19626901/146556875-a3068660-6fc1-4f5d-a19e-ab4c27eddcda.png)

_app.tsx
```typescript
import '../styles/globals.css'
import initAuth from '../service/auth/initAuth'
import type { AppProps } from 'next/app'

initAuth()
function MyApp({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />
}

export default MyApp
```

[api/logout.ts](https://gist.github.com/vpaladino778/13754f5827f7628b3368a7015e402ae3)
[api/login.ts](https://gist.github.com/vpaladino778/92dbe6191f31d237baf0249794ddeec9)
[login.tsx](https://gist.github.com/vpaladino778/86385ef171ecf440178f7298b728723c)
[initAuth.tsx](https://gist.github.com/vpaladino778/1bf0448958762d14ba82f98e49418309)

File Structure:
```
pages/
├─ api/
│  ├─ login.ts
│  ├─ logout.ts
├─ index.tsx
├─ _app.tsx
├─ dashboard.tsx
├─ login.tsx
service/auth/
├─ initAuth.tsx
```

**Comment by vpaladino778** [comment]
Here's my package.json
```json
{
  "name": "dashboard",
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
  "dependencies": {
    "@emotion/react": "^11.7.0",
    "@emotion/styled": "^11.6.0",
    "@mui/icons-material": "^5.2.1",
    "@mui/material": "^5.2.3",
    "dotenv": "^10.0.0",
    "firebase": "^9.6.1",
    "firebase-admin": "^10.0.1",
    "next": "^12.0.7",
    "next-firebase-auth": "^1.0.0-canary.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-firebaseui": "^6.0.0",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.0.4"
  },
  "devDependencies": {
    "@types/node": "16.11.12",
    "@types/react": "17.0.37",
    "eslint": "8.4.1",
    "eslint-config-next": "12.0.7",
    "typescript": "4.5.3"
  }
}
```

**Comment by kmjennison** [comment, marked as answer]
My suspicion is that you need to update how you're getting the auth app for Firebase v9 where you're passing it to Firebase UI. Remove Firebase compat, then rather than `firebaseAuth={firebase.auth()}`, use `firebaseAuth={getAuth(getApp())}`. See [v1.x example](https://github.com/gladly-team/next-firebase-auth/blob/v1.x/example/components/FirebaseAuth.js#L45).

**Reply by vpaladino778** [reply]
Thanks for you quick response, i really appreciate it. I made that modification, and modified the imports etc. I was able to open the login screen without seeing the error, but after logging in i get the following:

![image](https://user-images.githubusercontent.com/19626901/146585767-d16d151d-b200-4f8b-8d8b-236fb211e331.png)

Looks like the error happens on the login route this time, instead of upon opening initially. 

[api/login.ts](https://gist.github.com/vpaladino778/92dbe6191f31d237baf0249794ddeec9)

Console Output:
```
event - compiled client and server successfully (1003 modules)
error - FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).
error - FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).
wait  - compiling...
```

**Reply by kmjennison** [reply]
Can you log the API error [here](https://gist.github.com/vpaladino778/92dbe6191f31d237baf0249794ddeec9#file-login-ts-L11) to see what's going on?

**Reply by vpaladino778** [reply]
Whoops, should have included that the first time around:

Could the token be missing from the request header? It's also possible that this is just something wrong with my configuration. 
```
FirebaseAuthError: Firebase ID token has invalid signature. See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token.
    at FirebaseAuthError.FirebaseError [as constructor] (C:\Users\Vincent\Desktop\Repositories\dashboard\node_modules\firebase-admin\lib\utils\error.js:44:28)
    at FirebaseAuthError.PrefixedFirebaseError [as constructor] (C:\Users\Vincent\Desktop\Repositories\dashboard\node_modules\firebase-admin\lib\utils\error.js:90:28)
    at new FirebaseAuthError (C:\Users\Vincent\Desktop\Repositories\dashboard\node_modules\firebase-admin\lib\utils\error.js:149:16)
    at FirebaseTokenVerifier.mapJwtErrorToAuthError (C:\Users\Vincent\Desktop\Repositories\dashboard\node_modules\firebase-admin\lib\auth\token-verifier.js:237:20)
    at C:\Users\Vincent\Desktop\Repositories\dashboard\node_modules\firebase-admin\lib\auth\token-verifier.js:217:25
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  errorInfo: {
    code: 'auth/argument-error',
    message: 'Firebase ID token has invalid signature. See https://firebase.google.com/docs/auth/admin/verify-id-tokens for details on how to retrieve an ID token.'
  },
  codePrefix: 'auth'
}
```

**Reply by kmjennison** [reply]
Check out this troubleshooting note:
https://github.com/gladly-team/next-firebase-auth/tree/v1.x#i-get-an-authargument-error-with-message-firebase-id-token-has-invalid-signature

**Reply by vpaladino778** [reply]
Marking this as the answer, as it solved the problem related to my post. Thanks for all your help @kmjennison 

I'm still running into the `Firebase ID token has invalid signature` issue. I took a look at the trouble shooting guide. This is where i am pulling 'FIREBASE_CLIENT_EMAIL' from:

![image](https://user-images.githubusercontent.com/19626901/147128347-a4300636-ab00-4585-99c1-47835f1d4a18.png)
Does that look right?

This is what my `.env` looks like:

![image](https://user-images.githubusercontent.com/19626901/147129835-21cb95ba-7351-466f-b906-bdafe9939725.png)


Here's my initAuth.tsx: https://gist.github.com/vpaladino778/1020d0cb633d16cb5884ec50b7dc2697

Do I need the database URL to get this to work? Thanks again for all you're help. You should get a donation link going so I can buy you a coffee.

**Reply by kmjennison** [reply]
Yes, that should be the right email. I'm not sure what's going on, but I see you're using the Firebase emulator, which has caused problems for others: https://stackoverflow.com/a/68803294. Maybe try without the emulator if you haven't already?

Thanks for the kind words and appreciate the ☕ offer! But, no donation needed at this point—happy to be supporting open source when I can.

**Reply by vpaladino778** [reply]
That did the trick! I had to comment out `firebaseAuthEmulatorHost` from initAuth, and I had to remove the variable from the .env file.

I'll open a PR to add that suggestion to the 'Troubleshooting Guide'. Thanks again for your help.

**Comment by jonbattista** [comment]
@[kmjennison](https://github.com/kmjennison) Hello, I am trying to upgrade to Firebase 9.x as well. I am still calling `initAuth()` in `_app.js` but `getApp()` isnt returning the `next-firebase-auth` initialized app:
```
FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).
    at getApp (file:///Users/jon.battista/ltpro-frontend/node_modules/@firebase/app/dist/esm/index.esm2017.js:469:29)
    at eval (webpack-internal:///./pages/_app.js:36:65)
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  code: 'app/no-app',
  customData: { appName: '[DEFAULT]' },
  page: '/app'
}
```
I am using `"next": "12.0.7"`, `"next-firebase-auth": "^1.0.0-canary.7"`, and `"firebase": "^9.6.10"`

It should be initializing the app here: https://github.com/gladly-team/next-firebase-auth/blob/75bc40f0c041e9d71cf7292fdb940c05acbb6783/src/initFirebaseClientSDK.js#L13

and if I call `const firebaseApp = initializeApp(firebaseClientInitConfig);` right after `init()` everything works fine. So my config is valid, but `next-firebase-auth` isn't initializing the app

**Reply by kmjennison** [reply]
@jonbattista Could you try the example app in this repo to see if that works for you? Then, if you're still having trouble, please share a reproduction of the issue.
